### DISCUSSION TITLE
Image tag dynamically

### DISCUSSION THREAD

**Discussion by klinux** [woodpecker-ci/woodpecker#3457] [Ask a Question]
There any option to get image tag from secret? Because we needs to set image tag based on project settings, we have global pipelines and the pipeline is synced to the project, and we need to use image tag based on repository settings, there's any way to set this? WOODPECKER_ENVIRONMENT is not working, variables can not use from_secret.

Thank you

**Comment by zc-devs** [comment]
> we need to use image tag based on repository settings, there's any way to set this? 

Maybe [External Configuration API](https://woodpecker-ci.org/docs/next/administration/external-configuration-api) will suit you.

**Reply by tyler71** [reply]
Link is dead: https://woodpecker-ci.org/docs/administration/configuration/server#external-configuration-api

**Reply by zc-devs** [reply]
> image tag from secret

It won't be supported as it's visible in "Config" tab.

> we need to use image tag based on repository settings

Looks like you want repo variables. Read further #3661, #2330, #3723.
